zcash_primitives/src/transaction/components/tze.rs

```diff
@@ -0,0 +1,171 @@
+//! Structs representing the TZE components within Zcash transactions.
+#![cfg(feature = "zfuture")]
+
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+
+use std::io::{self, Read, Write};
+
+use std::convert::TryFrom;
+
+use crate::{
+    extensions::transparent as tze,
+    serialize::{CompactSize, Vector},
+};
+
+use super::amount::Amount;
+
+fn to_io_error(_: std::num::TryFromIntError) -> io::Error {
+    io::Error::new(io::ErrorKind::InvalidData, "value out of range")
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct TzeOutPoint {
```

Comment:
We should be naming this `OutPoint` and referring to it as `tze::OutPoint`, but we can do that refactor later.